x-pack/plugin/esql/qa/security/src/javaRestTest/resources/roles.yml

```diff
@@ -93,6 +93,21 @@ fls_user:
       field_security:
         grant: [ value ]
 
+fls_user2:
+  cluster: []
+  indices:
+    - names: [ 'lookup-user2' ]
+      privileges: [ 'read' ]
+      field_security:
+        grant: [ "org", "value" ]
+
+fls_user3:
+  cluster: []
+  indices:
+    - names: [ 'lookup-user2' ]
+      privileges: [ 'read' ]
+      field_security:
+        grant: [ "org", "value", "other" ]
```

Comment:
nit: the granted fields of `fls_userX` are always subsets of one another. Maybe it'd be interesting to have a `fls_user4` that has only `org` granted - then what happens if I have roles `fls_user` and `fls_user4`?

Reply:
Thanks Alex!
Added one more test with non overlapping roles